unikernel.ml

```diff
@@ -1384,12 +1366,39 @@ struct
                  ),
                (`Internal_server_error : H1.Status.t) ))
     | Ok () -> (
-        Utils.send_http_request http_client ~base_url:Builder_web.base_url
-          ~path:
-            ("/job/" ^ job ^ "/build/" ^ to_be_updated_unikernel
-           ^ "/main-binary")
+        let data_stream, push_chunks = Lwt_stream.create () in
+        let push () = Lwt_stream.get data_stream in
+        let url =
+          Builder_web.base_url ^ "/job/" ^ job ^ "/build/"
+          ^ to_be_updated_unikernel ^ "/main-binary"
+        in
+        let f _response _acc chunk = Lwt.return (push_chunks (Some chunk)) in
```

Comment:
We should probably check the response status here already.

Reply:
indeed..